openedx/openedx-platform
# feat: add more authentication information to swagger

## Description

* updates the `docs-settings` to make the generated swagger  `securityDefinitions` include both JWT and CSRF methods, as well as  basic. A few linter fixes happened as a side effect.
* Put in wordier descriptions for all three, since we don't have great shared documentation about authn/authz.
* Added CSRF to `login_session`, which also serves as a proof of concept  for other endpoints
* Also regenerated the swagger doc, which picked up some extra changes.

Generated swagger now has help and allows extra auth methods so some preveiously unusable endpoints can be hit.

## Testing instructions

* verify that `make swagger` works
* Plug the generated `lms-openapi.yml` into any OpenAPI  validator  or preview tool.  Online tools are available at https://swagger.io/, and most IDEs have validating plugins.
* Make sure that you can see the authentication methods and their description
* Make sure that you can see the authentication method CSRF on `login_session`

## Images

### general authentication modal
![the first page of the swagger authorizations modal, showing basic auth and JWT](https://github.com/user-attachments/assets/0e57eec9-2e94-4d0b-b8c9-563b060c2da6)
![The 2nd page showing CSRF](https://github.com/user-attachments/assets/0bbf3866-634f-4a66-86cb-88a5ee906c50)

### `login_session` authentication modal

![Modal for the login session endpoint, showing only CSRF](https://github.com/user-attachments/assets/5752b80d-f27c-4627-bf98-2b3c5fca5d2a)






FIXES: APER-3554
